<launch>

    <arg name="markerSize"          default="0.05"/>    <!-- in m -->
    <arg name="cam_id"              default="cam_1"/>
    <arg name="ref_frame"           default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement"   default="LINE" /> <!-- NONE, HARRIS, LINES, SUBPIX -->
    <arg name="marker_frame"        default="marker_frame"/>

    <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher">
        <remap from="/camera_info" to="/$(arg cam_id)/camera/rgb/camera_info" />
        <remap from="/image" to="/$(arg cam_id)/camera/rgb/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="$(arg cam_id)"/>
        <param name="marker_frame" value="$(arg marker_frame)" />
        <param name="corner_refinement" value="$(arg corner_refinement)" />
    </node>

</launch>

